phc-rails/app/controllers/api/v1/event_registrations_controller.rb

```diff
@@ -3,36 +3,33 @@ class Api::V1::EventRegistrationsController < ApplicationController
   before_action :verify_security_token
 
   def create
-    event_reg = EventRegistration.new
-
-    # Create account if salesforce id was not passed in
-    sf_id = params[:account_sfid]
-    if (sf_id.nil? || sf_id.empty?)
-      # Create account of the parameters passed in, sf_id will be nil
-      params[:sf_id] = params.delete :account_sfid
+    sf_id = params.delete :account_sfid
+    if sf_id.nil? || sf_id.empty? || sf_id = "\"\""
```

Comment:
Also, did you mean to put double equals signs (`sf_id == "\"\""`) on the last case?